From 6946cc8e181cd04dcced21a8b13b0bd31c5a4a8c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Mon, 11 Mar 2019 16:25:30 +0100
Subject: [PATCH] Add kernel configuration options for Turris 1.x
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is hacky approach and we should instead add these options to board
specific configuration.

Signed-off-by: Karel Kočí <karel.koci@nic.cz>
---
 target/linux/mpc85xx/config-4.14 | 69 ++++++++++++++++++++++++++++----
 1 file changed, 62 insertions(+), 7 deletions(-)

diff --git a/target/linux/mpc85xx/config-4.14 b/target/linux/mpc85xx/config-4.14
index 9d4170c85d..3f1a85aa12 100644
--- a/target/linux/mpc85xx/config-4.14
+++ b/target/linux/mpc85xx/config-4.14
@@ -42,6 +42,15 @@ CONFIG_BOOKE_WDT=y
 CONFIG_BOUNCE=y
 # CONFIG_BSC9131_RDB is not set
 # CONFIG_BSC9132_QDS is not set
+CONFIG_BTRFS_FS=y
+CONFIG_VFAT_FS=y
+CONFIG_FAT_DEFAULT_CODEPAGE=437
+CONFIG_FAT_DEFAULT_IOCHARSET="iso8859-1"
+CONFIG_NLS_CODEPAGE_437=y
+CONFIG_NLS_ISO8859_1=y
+# CONFIG_BTRFS_FS_CHECK_INTEGRITY is not set
+CONFIG_BTRFS_FS_POSIX_ACL=y
+CONFIG_BUILD_BIN2C=y
 # CONFIG_C293_PCIE is not set
 CONFIG_CLONE_BACKWARDS=y
 CONFIG_CLZ_TAB=y
@@ -59,6 +68,7 @@ CONFIG_CRYPTO_AKCIPHER2=y
 CONFIG_CRYPTO_AUTHENC=y
 CONFIG_CRYPTO_HASH=y
 CONFIG_CRYPTO_HASH2=y
+CONFIG_CRYPTO_LZO=y
 CONFIG_CRYPTO_HW=y
 CONFIG_CRYPTO_MANAGER=y
 CONFIG_CRYPTO_MANAGER2=y
@@ -72,6 +82,7 @@ CONFIG_CRYPTO_RSA=y
 # CONFIG_CRYPTO_SHA1_PPC_SPE is not set
 # CONFIG_CRYPTO_SHA256_PPC_SPE is not set
 CONFIG_CRYPTO_WORKQUEUE=y
+CONFIG_CRYPTO_XZ=y
 CONFIG_DEBUG_BUGVERBOSE=y
 # CONFIG_DEFAULT_UIMAGE is not set
 CONFIG_DNOTIFY=y
@@ -184,6 +195,10 @@ CONFIG_IRQ_FORCED_THREADING=y
 CONFIG_IRQ_WORK=y
 CONFIG_ISA_DMA_API=y
 CONFIG_KERNEL_GZIP=y
+CONFIG_JFFS2_FS_POSIX_ACL=y
+CONFIG_JFFS2_LZO=y
+CONFIG_JFFS2_RUBIN=y
+CONFIG_JFFS2_ZLIB=y
 CONFIG_KERNEL_START=0xc0000000
 # CONFIG_KSI8560 is not set
 CONFIG_LEGACY_PTYS=y
@@ -192,28 +207,53 @@ CONFIG_LIBFDT=y
 CONFIG_LOWMEM_CAM_NUM=3
 CONFIG_LOWMEM_SIZE=0x30000000
 CONFIG_LXT_PHY=y
-# CONFIG_MATH_EMULATION is not set
+CONFIG_LZO_COMPRESS=y
+CONFIG_LZO_DECOMPRESS=y
+CONFIG_MATH_EMULATION=y
+# CONFIG_MATH_EMULATION_FULL is not set
+CONFIG_MATH_EMULATION_HW_UNIMPLEMENTED=y
+CONFIG_MDIO_BOARDINFO=y
 CONFIG_MDIO_BUS=y
 CONFIG_MDIO_DEVICE=y
 CONFIG_MIGRATION=y
+CONFIG_MMC=y
+CONFIG_MMC_BLOCK=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_IO_ACCESSORS=y
+CONFIG_MMC_SDHCI_OF_ESDHC=y
+# CONFIG_MMC_SDHCI_PCI is not set
+CONFIG_MMC_SDHCI_PLTFM=y
+# CONFIG_MMC_TIFM_SD is not set
+# CONFIG_MMC_WBSD is not set
 # CONFIG_MMIO_NVRAM is not set
 CONFIG_MODULES_USE_ELF_RELA=y
 # CONFIG_MPC8536_DS is not set
 # CONFIG_MPC8540_ADS is not set
 # CONFIG_MPC8560_ADS is not set
-# CONFIG_MPC85xx_CDS is not set
+CONFIG_MPC85xx_CDS=y
 # CONFIG_MPC85xx_DS is not set
-# CONFIG_MPC85xx_MDS is not set
-# CONFIG_MPC85xx_RDB is not set
+CONFIG_MPC85xx_MDS=y
+CONFIG_MPC85xx_RDB=y
 CONFIG_MPIC=y
 # CONFIG_MPIC_MSGR is not set
 CONFIG_MPIC_TIMER=y
 # CONFIG_MPIC_U3_HT_IRQS is not set
 # CONFIG_MPIC_WEIRD is not set
 CONFIG_MPILIB=y
-# CONFIG_MTD_CFI is not set
+# CONFIG_MTD_CFI_NOSWAP is not set
+CONFIG_MTD_CFI_STAA=y
+CONFIG_MTD_JEDECPROBE=y
 CONFIG_MTD_M25P80=y
+CONFIG_MTD_RAM=y
+CONFIG_MTD_ROM=y
 CONFIG_MTD_SPI_NOR=y
+CONFIG_MTD_UBI=y
+CONFIG_MTD_UBI_BEB_LIMIT=20
+# CONFIG_MTD_UBI_BLOCK is not set
+# CONFIG_MTD_UBI_FASTMAP is not set
+# CONFIG_MTD_UBI_GLUEBI is not set
+CONFIG_MTD_UBI_WL_THRESHOLD=4096
+CONFIG_MUTEX_SPIN_ON_OWNER=y
 # CONFIG_MVME2500 is not set
 # CONFIG_NEED_DMA_MAP_STATE is not set
 # CONFIG_NEED_PER_CPU_EMBED_FIRST_CHUNK is not set
@@ -222,6 +262,7 @@ CONFIG_NEED_SG_DMA_LENGTH=y
 CONFIG_NLS=y
 # CONFIG_NONSTATIC_KERNEL is not set
 CONFIG_NO_BOOTMEM=y
+CONFIG_NR_CPUS=4
 CONFIG_NR_IRQS=512
 CONFIG_OF=y
 CONFIG_OF_ADDRESS=y
@@ -231,13 +272,14 @@ CONFIG_OF_FLATTREE=y
 CONFIG_OF_GPIO=y
 CONFIG_OF_IRQ=y
 CONFIG_OF_MDIO=y
+CONFIG_OF_MTD=y
 CONFIG_OF_NET=y
 CONFIG_OF_PCI=y
 CONFIG_OF_PCI_IRQ=y
 CONFIG_OF_RESERVED_MEM=y
 CONFIG_OLD_SIGACTION=y
 CONFIG_OLD_SIGSUSPEND=y
-# CONFIG_P1010_RDB is not set
+CONFIG_P1010_RDB=y
 # CONFIG_P1022_DS is not set
 # CONFIG_P1022_RDK is not set
 # CONFIG_P1023_RDB is not set
@@ -281,11 +323,12 @@ CONFIG_PPC_DOORBELL=y
 # CONFIG_PPC_EARLY_DEBUG is not set
 # CONFIG_PPC_EPAPR_HV_PIC is not set
 CONFIG_PPC_FSL_BOOK3E=y
-# CONFIG_PPC_I8259 is not set
+CONFIG_PPC_I8259=y
 # CONFIG_PPC_ICP_HV is not set
 # CONFIG_PPC_ICP_NATIVE is not set
 # CONFIG_PPC_ICS_RTAS is not set
 CONFIG_PPC_INDIRECT_PCI=y
+CONFIG_PPC_LIB_RHEAP=y
 CONFIG_PPC_MMU_NOHASH=y
 # CONFIG_PPC_MM_SLICES is not set
 # CONFIG_PPC_MPC106 is not set
@@ -331,6 +374,7 @@ CONFIG_SPE=y
 CONFIG_SPE_POSSIBLE=y
 CONFIG_SPI=y
 CONFIG_SPI_FSL_ESPI=y
+CONFIG_SPI_FSL_LIB=y
 CONFIG_SPI_MASTER=y
 CONFIG_SRCU=y
 # CONFIG_STRIP_ASM_SYMS is not set
@@ -349,7 +393,13 @@ CONFIG_TINY_SRCU=y
 # CONFIG_TQM8548 is not set
 # CONFIG_TQM8555 is not set
 # CONFIG_TQM8560 is not set
+CONFIG_TREE_RCU=y
 # CONFIG_TWR_P102x is not set
+CONFIG_UBIFS_FS=y
+CONFIG_UBIFS_FS_ADVANCED_COMPR=y
+CONFIG_UBIFS_FS_LZO=y
+CONFIG_UBIFS_FS_XZ=y
+CONFIG_UBIFS_FS_ZLIB=y
 CONFIG_UCC=y
 CONFIG_UCC_FAST=y
 CONFIG_UCC_GETH=y
@@ -368,8 +418,12 @@ CONFIG_VM_EVENT_COUNTERS=y
 CONFIG_WATCHDOG_CORE=y
 # CONFIG_XES_MPC85xx is not set
 # CONFIG_XPS_USB_HCD_XILINX is not set
+CONFIG_XOR_BLOCKS=y
+CONFIG_XPS=y
 CONFIG_XZ_DEC_BCJ=y
 CONFIG_XZ_DEC_POWERPC=y
+CONFIG_ZLIB_DEFLATE=y
+CONFIG_ZLIB_INFLATE=y
 CONFIG_FB_FSL_DIU=n
 CONFIG_VGA_CONSOLE=y
 CONFIG_VGACON_SOFT_SCROLLBACK=n
@@ -414,6 +468,7 @@ CONFIG_NFS_FS=y
 CONFIG_NFS_V2=y
 CONFIG_NFS_V3=y
 CONFIG_ROOT_NFS=y
+CONFIG_SPI_SPIDEV=y
 CONFIG_WATCHDOG_PRETIMEOUT_GOV=y
 CONFIG_WATCHDOG_PRETIMEOUT_GOV_NOOP=n
 CONFIG_WATCHDOG_PRETIMEOUT_DEFAULT_GOV_NOOP=n
-- 
2.21.0

